### ISSUE TITLE
Authorize query null for persited query

### ISSUE THREAD

**Issue by TheNoobProgrammeur** [graphql-java/graphql-java#2636]
Hi ! I want to use persisted query.

To respect apollo persited query protocol 
(https://www.apollographql.com/docs/react/api/link/persisted-queries/#protocol).

I will send one request where my query is not defined.
But ExecutionInput builder don't support null query.
https://github.com/graphql-java/graphql-java/blob/abbb85644ef513631ed4fe3c3878024a63fa8595/src/main/java/graphql/ExecutionInput.java#L224

In graphql kickstart spring boot lib
One exception is return when `GraphQLSingleInvocationInput.createExecutionInput()` is executed.
I see in this MR your raise a question but I don't find a solution.

https://github.com/graphql-java/graphql-java/pull/2013#issuecomment-786458259

It is posibel to overide the ExecutionInput build rule ?

**Comment by bbakerman** [comment]
Yes it cant be null BUT it can be a marker that helps you know its a persisted query

See `graphql.execution.preparsed.persisted.ApolloPersistedQuerySupportTest` for an example
